base/essentials.jl

```diff
@@ -435,6 +435,19 @@ julia> reinterpret(Float32, UInt32[1 2 3 4 5])
 1×5 reinterpret(Float32, ::Matrix{UInt32}):
  1.0f-45  3.0f-45  4.0f-45  6.0f-45  7.0f-45
 ```
+
+Additionally, `reinterpret` does not require the array `A` to have a dense block of memory;
+it can be "lazy" arrays. For instance, `reinterpret` on range `1:6` works the same as the
+dense vector `collect(1:6)`:
+
+```jldoctest
+julia> reinterpret(Complex{Int}, 1:6) # `UnitRange` does not have a buffer holding array values
+3-element reinterpret(Complex{$Int}, ::UnitRange{$Int}):
```

Comment:
Doctests are only running and passing on 64-bit so perhaps
```suggestion
3-element reinterpret(Complex{Int64, ::UnitRange{Int64}):
```
and likewise below.

Reply:
Good to know!

Reply:
Hmm, that sounds like a step backwards to me. Why wouldn't you write it in a way that will pass on any platform?